File: frontend/src/services/adminRequestService.js
import springapi from './springapi'

const SPRING_API = "/adminrequest";

export const createAdminRequest = async (
    email,
    fullname,
    title,
    chatbot_name,
    department_program,
    purpose
) => {
    const response = await springapi.post(`${SPRING_API}`, {
        email,
        fullname,
        title,
        chatbot_name,
        department_program,
        purpose
    });
    return response.data;
}


export const getAllRequest = async(status) =>{
    const response = await springapi.get(`${SPRING_API}`,{
        params: {status}
    });
    return response.data;
}

export const approveRequest = async(request_id) =>{
    const response = await springapi.post(`${SPRING_API}/approverequest`, null,{
        params: {request_id}
    })
    return response.data;
}

export const rejectRequest = async(request_id) =>{
    const response = await springapi.post(`${SPRING_API}/rejectrequest`, null, {
        params: {request_id}
    })
    return response.data;
}

File: spring-service/src/main/java/com/acadprobot/admin/service/AdminRequestService.java
package com.acadprobot.admin.service;

import com.acadprobot.admin.model.AdminChatbotRequest;
import com.acadprobot.admin.repository.AdminChatbotRequestRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Service
public class AdminRequestService {

    @Autowired
    private AdminChatbotRequestRepository adminChatbotRequestRepository;

//    create
    public AdminChatbotRequest createAdminChatbotRequest(
            String email,
            String fullname,
            String title,
            String chatbot_name,
            String departmentProgram,
            String purpose
    ){
        AdminChatbotRequest adminChatbotRequest = new AdminChatbotRequest();
        adminChatbotRequest.setEmail(email);
        adminChatbotRequest.setFullname(fullname);
        adminChatbotRequest.setTitle(title);
        adminChatbotRequest.setChatbot_name(chatbot_name);
        adminChatbotRequest.setDepartment_program(departmentProgram);
        adminChatbotRequest.setPurpose(purpose);

        return adminChatbotRequestRepository.save(adminChatbotRequest);
    }

//    read
    public List<AdminChatbotRequest> getAllRequestByStatus(String status){
        return adminChatbotRequestRepository.findByStatus(status);
    }

//    update
    public String approveRequest(UUID id){
        Optional<AdminChatbotRequest> requestOpt = adminChatbotRequestRepository.findById(id);
        if(requestOpt.isEmpty()){
            return "Request not found" + id;
        }
        AdminChatbotRequest adminChatbotRequest = requestOpt.get();
        adminChatbotRequest.setStatus("approved");

        adminChatbotRequestRepository.save(adminChatbotRequest);

        return "Request approved for " + adminChatbotRequest.getEmail();
    }

    public String rejectRequest(UUID id){
        Optional<AdminChatbotRequest> requestOpt = adminChatbotRequestRepository.findById(id);
        if(requestOpt.isEmpty()){
            return "Request not found" + id;
        }
        AdminChatbotRequest adminChatbotRequest = requestOpt.get();
        adminChatbotRequest.setStatus("rejected");

        adminChatbotRequestRepository.save(adminChatbotRequest);

        return "Request rejected for " + adminChatbotRequest.getEmail();
    }
}
